db/queries/assessment_records/_helpers.py

```diff
@@ -1,79 +1,59 @@
-    ),
-}
-
-
-def get_mapper(application_type: str) -> dict:
-    """get_mapper A factory returning the dictionary associated with the given
-    `application_type`.
-
-    :param application_type: An application type for which we have an
-    associated mapper dict. For example, "cof" is associated with
-    `cof_json_mapper` through this function.
-    :raises KeyError: Raised if the given `application_type` doesn't have an
-    associated dict.
-    :return: A dictionary with keys equal to the column names in `db.models.
-    AssessmentRecord` and values equal to the jsonpath which will extract the
-    column values from a json with the corresponding application type.
-    """
-
-    match application_type:
-
-        case "COF":
-            return cof_json_mapper
-
-        case _:
-            raise KeyError("Valid application type not given.")
-
-
-def derive_values_from_json(loaded_json: dict, application_type: str) -> dict:
-    """derive_values_from_json Given a loaded json dictionary `loaded_json` and
-    an `application_type` we extract key values from the json and return them
-    as a dictionary.
-
-    :param loaded_json: An application json which has been converted to a
-    python `dict` object.
-    :param application_type: A string matching the cases found in `get_mapper`
-    :return: A dictionary containing the values extracted from `loaded_json`.
-    Keys match the keys in the associated mapping dict, such as
-    `cof_json_mapper`.
-    """
-
-    mapper = get_mapper(application_type)
-
-    # Uses a CTE query to extract values from blob, faster then using
-    # python jsonpath libraries.
-
-    loaded_blob = select(cast(loaded_json, JSONB).label("json_value")).cte(
-        "blob"
-    )
-
-    selects = [
-        func.jsonb_path_query_first(loaded_blob.c.json_value, json_path).label(
-            column_name
+import jsonpath_rw_ext
+
+
+def derive_application_values(application_json):
+    # TODO: implement mapping function to match
+    #  fund+round fields to derived values
+    derived_values = {}
+    application_id = application_json["id"]
+    print(f"deriving values for application id: {application_id}.")
+    try:
+        asset_type = (
+            jsonpath_rw_ext.parse(
+                '$.forms[*].questions[*].fields[?(@.key == "yaQoxU")]'
+            )
+            .find(application_json)[0]
+            .value["answer"]
         )
-        for column_name, json_path in mapper.items()
-    ]
-
-    extract_fields_stmt = select(*selects).select_from(loaded_blob)
+    except Exception:
+        print(
+            f"Could not extract asset_type from application: {application_id}."
+        )
+        asset_type = "Not asset type specified."
+    try:
+        funding_one = (
+            jsonpath_rw_ext.parse(
+                '$.forms[*].questions[*].fields[?(@.key =="JzWvhj")]'
+            )
+            .find(application_json)[0]
+            .value["answer"]
+        )
+    except Exception:
```

Comment:
Will this definitely throw an exception or will it return `None`?

Reply:
no in the event of a null in the json, it throws an error. I have changed this to set the value as 0 👍